replica/database.hh

```diff
@@ -409,7 +409,7 @@ private:
     sstables::compaction_strategy _compaction_strategy;
     // TODO: Still holds a single compaction group, meaning all sstables are eligible to be compacted with one another. Soon, a table
     //  will be able to hold more than one group.
-    std::unique_ptr<compaction_group> _compaction_group;
+    lw_shared_ptr<compaction_group> _compaction_group;
```

Comment:
Can't we use regular pointers? I'd like to move away from the laziness of using shared pointers.

For static compaction groups, it shouldn't matter.

For dynamic compaction groups, we can use a barrier like @bhalevy suggests.